r_d19/DynamicIntArray.cpp

```diff
@@ -0,0 +1,111 @@
+#include "DynamicIntArray.h"
+
+DynamicIntArray::DynamicIntArray()
+{
+	this->array = nullptr;
+	this->size = 0;
+}
+
+DynamicIntArray::DynamicIntArray(std::size_t size)
+{
+	this->array = new int[size];
+	this->size = size;
+}
+
+DynamicIntArray::DynamicIntArray(const DynamicIntArray& other)
+{
+	std::size_t originalSize = other.getSize();
+	this->array = new int[originalSize];
+	this->size = originalSize;
+
+
+	for (std::size_t i = 0; i < originalSize; i++) {
+		this->array[i] = other[i];
+	}
+}
+
+DynamicIntArray::~DynamicIntArray()
+{
+	delete[] this->array;
+}
+
+DynamicIntArray& DynamicIntArray::operator=(const DynamicIntArray& other)
+{
+	if (this == &other) {
+		return *this;
+	}
+
+	std::size_t originalSize = other.getSize();
+	this->array = new int[originalSize];
+	this->size = originalSize;
+
+
+	for (std::size_t i = 0; i < originalSize; i++) {
+		this->array[i] = other[i];
+	}
+
+	return *this;
+}
+
+int& DynamicIntArray::operator[](std::size_t index)
+{
+	return this->array[index];
+}
+
+const int& DynamicIntArray::operator[](std::size_t index) const
+{
+	return this->array[index];
+}
+
+void DynamicIntArray::setSize(std::size_t newSize)
+{
+	delete[] this->array;
+
+	this->size = newSize;
+	this->array = new int[newSize];
```

Comment:
You should save old content if possible not just remove it as it wasn't there at all